lib/stellar/client.rb

```diff
@@ -196,6 +196,59 @@ def change_trust(
       horizon.transactions._post(tx: envelope_base64)
     end
 
+    Contract(C::KeywordArgs[
+      tx_envelope: Stellar::TransactionEnvelope,
+      options: Maybe[{ skip_memo_required_check: C::Bool }]
+    ] => Any)
+    def submit_transaction(tx_envelope:, options: { skip_memo_required_check: false })
+      if !options[:skip_memo_required_check]
+        check_memo_required(tx_envelope)
+      end
+      @horizon.transactions._post(tx: tx_envelope.to_xdr(:base64))
+    end
+
+    Contract Stellar::TransactionEnvelope => Any
+    def check_memo_required(tx_envelope)
+      tx = tx_envelope.tx
+      # Check transactions where the .memo field is nil or of type MemoType.memo_none
+      if !tx.memo.nil? && tx.memo.type != Stellar::MemoType.memo_none
+        return
+      end
+      destinations = Set.new
+      tx.operations.each do |op|
+        if op.body.type == Stellar::OperationType.payment
+          destination = op.body.value.destination
+        elsif op.body.type == Stellar::OperationType.path_payment_strict_receive
+          destination = op.body.value.destination
+        elsif op.body.type == Stellar::OperationType.path_payment_strict_send
+          destination = op.body.value.destination
+        elsif op.body.type == Stellar::OperationType.account_merge
+          # There is no AccountMergeOp, op.body is an Operation object
+          # and op.body.value is a PublicKey (or AccountID) object.
+          destination = op.body.value
+        else
+          return
+        end
+        
+        if destinations.include?(destination)
+          next
+        end
+        destinations.add(destination)
+       
+        kp = Stellar::KeyPair.from_public_key(destination.value)
+        begin
+          info = account_info(kp.address)
+        rescue Faraday::ResourceNotFound
+          # Don't raise an error if its a 404, but throw one otherwise
+          next
+        end
+        if info.data["config.memo_required"] == "MQ=="
+          # MQ== is XDR for "1"
```

Comment:
This is base64, not xdr.
```suggestion
          # MQ== is the base64 encoded string for the string "1"
```